### ISSUE TITLE
Unable to install: This app cannot be enabled because it makes the server unstable 

### ISSUE THREAD

**Issue by debugger24** [nextcloud/user_oidc#678]
Unable to install: This app cannot be enabled because it makes the server unstable

Nextcloud Version: 27.0.2
user oidc Version: 1.3.3

Where can I find more details about the error ?

<img width="1255" alt="image" src="https://github.com/nextcloud/user_oidc/assets/12707743/55804aeb-9271-43e4-ba08-3861250a4cb0">

**Comment by ghostklart** [comment]
having the same issue with:
nextcloud (docker): 28.0.4
open id version: 5.0.2

**Comment by krair** [comment]
Same issue here. Nothing in the logs to guide me as to why.

Currently on the `production-fpm` tag which puts me at 28.0.5 and trying to use version 5.0.2 of this app. Any tips?

**Comment by julien-nc** [comment]
@ghostklart @krair Have you tried installing/enabling another app? I think this is a NC issue rather than a problem with this specific app.

**Comment by krair** [comment]
@julien-nc You're correct. It appears I can't install other apps as well. Any tips on where to poke around? I found this: https://github.com/nextcloud/server/issues/45378 or is it something else? Perhaps upgrading to 28.0.6 will fix the issue...

**Comment by ghostklart** [comment]
@julien-nc I started from scratch and changed docker installation from linuxserver to official nextcloud, and storage from local mount to minio. Now working fine. So far, seems like issue was with storage permissions. But no exact errors in logs.

**Comment by krair** [comment]
Upgraded to 28.0.6. Still the same issue. Investigating further...

**Comment by krair** [comment]
Found the issue, it's working now. Apparently the bug has been fixed, but obviously still affecting me. https://github.com/nextcloud/server/issues/44685

I created the temporary redirect for Traefik (my reverse proxy).

**Comment by susnux** [comment]
This is not a bug but a faulty configuration of your reverse proxy.

Nextcloud is running on a webserver like Apache, if Apache is configure with `DirectorySlash` it will emit a redirect from `some/url` to `some/url/`.
This is done **before** Nextcloud can intercept the request.
If you just run Apache, this would be ok as the browser then would request `some/url/` and Apache simply pass this to Nextcloud.

*But* if you run a reverse proxy on HTTPS that passes the requests to Apache using HTTP, then Apache will emit the redirect using HTTP (no "S"!).
So **your reverse proxy** has to **rewrite** the redirect (`Location`) header from HTTP to HTTPS.

NGinx does this by default (in most cases), see [proxy pass](http://nginx.org/en/docs/http/ngx_http_proxy_module.html?&_ga=1.74997266.187384914.1443061481#proxy_pass)
But e.g. Traefik does not do this, so you need to adjust your config.

**Comment by githubbbie** [comment]
Hmmm, running 29.0.1.1 and getting the error upon trying to enable apps:

Error: This app cannot be enabled because it makes the server unstable

![image](https://github.com/user-attachments/assets/bd461bde-5eb6-49c0-abc6-c35d55de9ac3)

```
$ cat /etc/lsb-release
DISTRIB_ID=Pop
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Pop!_OS 22.04 LTS"
```

Attaching my docker-compose.yml:
[docker-compose.yml.txt](https://github.com/user-attachments/files/16714365/docker-compose.yml.txt)

(remove the .txt to test)

My next step is to downgrade to version 28 and try to get Nextcloud running.

BTW, I am running this on localhost HTTP without any reverse proxy. Running: localhost:8080

**Comment by githubbbie** [comment]
Ok, confirm this bug is fixed in v28.0.9.1 with this docker.compose.yml:

[docker-compose.yml.txt](https://github.com/user-attachments/files/16715081/docker-compose.yml.txt)

**Comment by githubbbie** [comment]
Consider rolling back the v29 series as STABLE or PRODUCTION ready.

**Comment by susnux** [comment]
> Ok, confirm this bug is fixed in v28.0.9.1 with this docker.compose.yml:

Please note that you are using the Community maintained docker image, so please report that issue here:
https://github.com/nextcloud/docker

The only officially supported installation method for docker is the all-in-one docker image:
https://github.com/nextcloud/all-in-one
